pkg/controller/controller.go

```diff
@@ -288,6 +288,13 @@ func (c *Controller) syncCluster(rediscluster *rapi.RedisCluster) (forceRequeue
 		return forceRequeue, err
 	}
 
+	for _, p := range redisClusterPods {
```

Comment:
the logic is correct, but the `redisClusterPods` slice needs to be filtered of the pods that were deleted if `Status.Reason == NodeLost".  Else the function `NewRedisAdmin(...)` line 299 will fail to create the connections with all Redis process.
Adding a temporary Slide of the valid Pods can a solution.